.github/workflows/benchmark_execution.yml

```
@@ -67,12 +67,16 @@ jobs:
         run: ./build_tools/scripts/git/update_runtime_submodules.sh
       - name: "Generating benchmark matrix"
         id: generate
-        run: |
-          gcloud storage cp "${BENCHMARK_CONFIG_GCS_ARTIFACT}" "${BENCHMARK_CONFIG}"
-          echo benchmark-matrix=$(jq \
-              'to_entries | map({"device_name": .key, "host_environment": .value.host_environment})' \
-              "${BENCHMARK_CONFIG}") \
-            >> "${GITHUB_OUTPUT}"
+        run: >
+          gcloud storage cp "${BENCHMARK_CONFIG_GCS_ARTIFACT}" "${BENCHMARK_CONFIG}";
+          echo benchmark-matrix=$(jq 'to_entries[]
+          | .key as $device_name
+          | .value.host_environment as $host_environment
+          | (.value.shards | length) as $shard_count
+          | .value.shards[]
+          | { device_name: $device_name, host_environment: $host_environment,
+          shard: { index: .shard_index, count: $shard_count }
+          }' "${BENCHMARK_CONFIG}" | jq -s '.') >> "${GITHUB_OUTPUT}"
```

Seems like it's possible to write with a single `jq` call
```suggestion
          echo benchmark-matrix=$(jq 'to_entries
            | map(
               .key as $device_name
              | .value.host_environment as $host_environment
              | (.value.shards | length) as $shard_count
              | .value.shards[]
              | { device_name: $device_name, host_environment: $host_environment,
                shard: { index: .shard_index, count: $shard_count }
            )}' "${BENCHMARK_CONFIG}") >> "${GITHUB_OUTPUT}"
```

I don't have a strong opinion on which on is better : )

---

Yes, works. I also have no strong opinion so I adopted yours.